keyboards/atreus/keymaps/ibnuda/keymap.c

```diff
@@ -0,0 +1,281 @@
+#include QMK_KEYBOARD_H
+
+typedef enum {
+    SINGLE_TAP,
+    SINGLE_HOLD,
+    DOUBLE_TAP,
+} td_state_t;
+
+static td_state_t td_state;
+
+int current_dance(qk_tap_dance_state_t *state);
+
+void dance_tmb_finished(qk_tap_dance_state_t *state, void *user_data);
+void dance_tmb_reset(qk_tap_dance_state_t *state, void *user_data);
+
+// enum for tap dances.
+enum {
+    TD_DLT_CTLDLT = 0,
+    TD_SCLN_CLN,
+    TD_LEFT_THUMB,
+};
+
+// enum for combos.
+enum combos {
+    // left hand combinations.
+    COLON_COMMA,
+    COMMA_DOT,
+    DOT_P,
+    Q_J,
+    J_K,
+
+    // right hand combinations.
+    L_R,
+    R_C,
+    C_G,
+    V_W,
+    W_M,
+
+    // both hands combinations.
+    DOT_C,
+    J_W,
+    P_G,
+    U_H,
+    K_M,
+};
+
+#define _BASE    0
+#define _LOWER   1
+#define _RAISE   2
+#define _ADJUST  3
+#define _MUIS    4
+
+#define ____ KC_TRNS
+
+// short.
+#define KC_QUES LSFT(KC_SLSH)
+#define KC_UNDR LSFT(KC_MINS)
+#define MU_UP   KC_MS_UP
+#define MU_DN   KC_MS_DOWN
+#define MU_LFT  KC_MS_LEFT
+#define MU_RGHT KC_MS_RIGHT
+#define MU_BTN1 KC_MS_BTN1
+#define MU_BTN2 KC_MS_BTN2
+#define MU_WD   KC_MS_WH_DOWN
+#define MU_WU   KC_MS_WH_UP
```

Comment:
QMK already has short-form aliases for all of these keycodes; you don't need to define any of these if you use the pre-built codes.

Please see:
- [US ANSI Shifted Keys - QMK Firmware Docs](https://docs.qmk.fm/#/keycodes_us_ansi_shifted)
- [Mouse Keys - QMK Firmware Docs](https://docs.qmk.fm/#/feature_mouse_keys?id=mapping-mouse-actions)